### PR TITLE
#7Task. tanya.shulha

### PR DESCRIPTION
**Pull request by tanyashulha**
https://travis-ci.com/github/tanyashulha/js-assignments/builds/175206330

### CODE REVIEW

File: task/07-yield-tasks.js [hunk 1]
```diff
@@ -47,7 +63,18 @@ function* get99BottlesOfBeer() {
...
+    
+    function getSequence(n) {
+        if(num[n] || n === 0){
+            return num[n];
+        }
+        num.push(getSequence(n - 1) + getSequence( n - 2));
+        return num[n]
+    }
+    for(let i = 0; i < 40; i++){
+        yield getSequence(i)
```
Comment (AlexMokin) [inline task/07-yield-tasks.js]: do it without recursion and assume you need to produce an infinite sequence

File: task/07-yield-tasks.js [hunk 2]
```diff
@@ -145,7 +210,17 @@ function* mergeSortedSequences(source1, source2) {
  *   Most popular implementation of the logic in npm https://www.npmjs.com/package/co
  */
 function async(generator) {
-    throw new Error('Not implemented');
+    let newGenerator = generator();
+
+    return Promise.resolve((function func(val){
```
Comment (AlexMokin) [inline task/07-yield-tasks.js]: split the definition of the func function and the usage of this function. Use better name for "func"

**Review by AlexMokin — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

File: task/07-yield-tasks.js [hunk 3]
```diff
@@ -47,9 +63,19 @@ function* get99BottlesOfBeer() {
  *
  */
 function* getFibonacciSequence() {
-    throw new Error('Not implemented');
-}
+    let num = [0, 1];
+    let n;
 
+    for(let i = 0; i < num.length; i++) {
```
Comment (AlexMokin) [inline task/07-yield-tasks.js]: this for-loop doesn't do anything useful. The while-loop below is infinite meaning `i` is always 0

**Review by AlexMokin — COMMENTED** [review]
Please check a new comment

I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

File: task/07-yield-tasks.js (hunk 3 above)
Comment (tanyashulha) [reply task/07-yield-tasks.js]: yes, you are right. thank you!

**Comment by IlyaMokin** [comment]
Looks good, thanks! closing the PR as accepted

**Pull request closed without merging**